tests/test_flows.py

```diff
@@ -1246,14 +1249,25 @@ def test_higher_layer_data_payload_trigger(self):
         mid = env.controller.dev_send_1905(env.agents[0].mac, 0x8018,
                                            tlv(0xA0, 0x04b1, "{0x00 %s}" % payload))
 
-        debug("Confirming higher layer data message was received in the agent")
+        debug("Confirming higher layer data message was received in one of the agent's radios")
 
-        self.check_log(env.agents[0], r"HIGHER_LAYER_DATA_MESSAGE")
+        received_in_radio0, _, _ = self.check_log(env.agents[0].radios[0],
+                                                  r"HIGHER_LAYER_DATA_MESSAGE",
+                                                  fail_when_not_found=False)
+        received_in_radio1, _, _ = self.check_log(env.agents[0].radios[1],
+                                                  r"HIGHER_LAYER_DATA_MESSAGE",
+                                                  fail_when_not_found=False)
+        if (received_in_radio0 == received_in_radio1):
+            self.fail(f"higher layer data message was received and acknowledged by "
+                      f"{int(received_in_radio0) + int(received_in_radio1)} agent's radios, "
+                      f"expected only 1")
```

Comment:
```suggestion
                      f"expected exactly 1")
```
1. Since you decided to include in the same condition:
false,false
and
true,true
then your comment should reflect that.

2. This is a test so I would consider the following comment a nitpick - but:
I prefer the condition and the error-condition to be pre-acquired to the point you use them.
Meaning:
- Calculate the number of agents that received the message
- Then, compare it against the expected value.
- Then, print the expected and actual as part of the error log print

Reply:
suggestion applied